### ISSUE TITLE
CMake error "Source directory is polluted" with zxing C++ port, CMake 3.20.2 and MSVS Professional 2019 Version 16.10.3

### ISSUE THREAD

**Issue by VolkerBartheld** [zxing/zxing#1432]
When running CMake 3.20.2 to build a MSVS Professional 2019 Version 16.10.3 solution for  https://github.com/zxing/zxing/tree/00f634024ceeee591f54e6984ea7dd666fab22ae/cpp ,you're getting the following error:

```
CMake Error at CMakeLists.txt:13 (message):
  Source directory is polluted:

    * remove CMakeCache.txt
    * remove CMakeFiles directory

-- Configuring incomplete, errors occurred!
See also "D:/Repositories/zxing/cpp/build/CMakeFiles/CMakeOutput.log".

```
I suggest removing the entire section saying
```
# Check for polluted source tree.
if(EXISTS ${CMAKE_SOURCE_DIR}/CMakeCache.txt OR
    EXISTS ${CMAKE_SOURCE_DIR}/CMakeFiles)
    message(FATAL_ERROR
        "Source directory is polluted:"
        "\n  * remove CMakeCache.txt"
        "\n  * remove CMakeFiles directory")
endif()

```
in https://github.com/zxing/zxing/blob/00f634024ceeee591f54e6984ea7dd666fab22ae/cpp/CMakeLists.txt.

Afterwards the .sln will be created OK and build without errors.

**Comment by srowen** [comment]
This code doesn't exist anymore but fair comment for anyone using an old version when it existed

**Comment by VolkerBartheld** [comment]
@srowen: Thanks for getting back to me so quickly. Yes, I have noticed that the master branch does not have the C++ port in anymore, which is quite a bummer, since there does not seem to be any usefule QR code reader application for Windows platforms out there anymore. I tried for hours and they all suck. So I decided to take the reference tool i. e.build zxing myself and ran across the issue.

There is also another little problem with CRLF/<< endl being sent after decoding a QR code (see https://github.com/zxing/zxing/issues/1433) that was spoiling my fun when working with these COVID vaccination certificates the EU is now handing out to people who are fully vaccinated. This is essentially a binary data blob (encrypted/signed) and you don't want a CRLF being appended to that data.

I do not know where to put that information, so the next best idea was to raise a bug.

Thanks for your consideration and keep up the good work!
